UnityAds/src/main/java/com/mopub/mobileads/UnityRewardedVideo.java

```diff
@@ -1,248 +1,69 @@
-
-        setAutomaticImpressionAndClickTracking(false);
-
-        UnityAds.load(UnityRouter.placementIdForServerExtras(adData.getExtras(), ""), mUnityLoadListener);
-    }
-
-    /**
-     * IUnityAdsLoadListener instance. Contains ad load success and fail logic.
-     */
-    private IUnityAdsLoadListener mUnityLoadListener = new IUnityAdsLoadListener() {
-        @Override
-        public void onUnityAdsAdLoaded(String placementId) {
-            mPlacementId = placementId;
-
-            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity rewarded video successfully loaded for placementId " + placementId);
-            MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
-
-            if (mLoadListener != null) {
-                mLoadListener.onAdLoaded();
-            }
-        }
-
-        @Override
-        public void onUnityAdsFailedToLoad(String placementId, UnityAdsLoadError error, String message) {
-            mPlacementId = placementId;
-
-            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity rewarded video failed to load for placement " +
-                    placementId + ", with error message: " + message);
-            MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, MoPubErrorCode.NETWORK_NO_FILL.getIntCode(), MoPubErrorCode.NETWORK_NO_FILL);
-
-            if (mLoadListener != null) {
-                mLoadListener.onAdLoadFailed(MoPubErrorCode.NETWORK_NO_FILL);
-            }
-        }
-    };
-
     @Override
-    public void show() {
-        MoPubLog.log(SHOW_ATTEMPTED, ADAPTER_NAME);
-
-        if (mLauncherActivity == null) {
-            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Failed to show Unity rewarded video as the activity calling it is null.");
-            MoPubLog.log(SHOW_FAILED, ADAPTER_NAME,
-                    MoPubErrorCode.VIDEO_PLAYBACK_ERROR.getIntCode(),
-                    MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
-
-            if (mInteractionListener != null) {
-                mInteractionListener.onAdFailed(MoPubErrorCode.VIDEO_PLAYBACK_ERROR);
-            }
-            return;
-        }
-
-        if (mPlacementId == null) {
-            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity Ads received call to show before successfully loading an ad");
-        }
-
-        MediationMetaData metadata = new MediationMetaData(mLauncherActivity);
-        metadata.setOrdinal(++impressionOrdinal);
-        metadata.commit();
-
-        UnityAds.show(mLauncherActivity, mPlacementId, mUnityShowListener);
-    }
-
-    /**
-     * IUnityAdsShowListener instance. Contains logic for callbacks when showing ads.
-     */
-    private IUnityAdsShowListener mUnityShowListener = new IUnityAdsShowListener() {
-
-        @Override
-        public void onUnityAdsShowStart(String placementId) {
-            if (mInteractionListener != null) {
-                mInteractionListener.onAdShown();
-                mInteractionListener.onAdImpression();
+    protected IUnityAdsShowListener getIUnityAdsShowListener() {
```

Comment:
I guess we can keep it for future improvement but we could reuse some of the show listener implementation and have `UnityVideoAd` implement `IUnityShowListener` (to reuse since behaviour is almost same for Rewarded and Interstitial)